src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -78,52 +76,19 @@ case class Instance(
   override def region: Option[String] = agentInfo.flatMap(_.region)
 
   /**
-    * Factory method for creating provisioned instance from Scheduled instance for apps
+    * Factory method for creating provisioned instance from Scheduled instance
     * @return new instance in a provisioned state
     */
-  def provisioned(agentInfo: Instance.AgentInfo, networkInfo: core.task.state.NetworkInfo, app: AppDefinition, now: Timestamp, taskId: Task.Id): Instance = {
-    require(isScheduled, s"Instance '${instanceId}' must not be in state '${state.condition}'. Scheduled instance is required to create provisioned instance.")
+  def provisioned(agentInfo: Instance.AgentInfo, runSpecVersion: Timestamp, tasks: Seq[Task], now: Timestamp): Instance = {
```

Comment:
Hm, how about passing a `Map[Task.Id, Task]` instead of `Seq[Task]`?

Reply:
Maybe we should call the param provisionalTasks?

Reply:
why Map? the fact that instance needs to have it as a id indexed map is an implementation detail to me. Someone has to do the seq to map transformation, I am fine with the updater knowing the instance internals does it, not every client wanting to submit the update...

Reply:
Yup. Fine with me.